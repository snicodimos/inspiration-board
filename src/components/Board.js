import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';

class Board extends Component {
  constructor() {
    super();

    this.state = {
      cards: []
    };
  }
  componentDidMount() {
    axios.get('https://inspiration-board.herokuapp.com/boards/Semret/cards')
    .then((response) => {
      this.setState({
        cards: response.data
      });
    })
    .catch((error) => {
      this.setState({
        error: error.message
      });
    })
  }
  deleteThisCard = (id) => {
    console.log("a specific card is being deleted");
    const updatedCards = this.state.cards
    console.log(updatedCards.length);
    axios.delete(`https://inspiration-board.herokuapp.com/cards/${id}`)
    .then((response) => {

      const filteredCards = updatedCards.filter(({card}) => card.id !== id )
      console.log(filteredCards.length);
      this.setState({
        cards: filteredCards
      });
    })
    .catch((error) => {
      this.setState({
        error: error.message
      });
    })
  }

  addNewCard = (newCard) => {
    console.log("adding new card", newCard);
    const updateCards = [...this.state.cards, newCard]

    axios.post(`https://inspiration-board.herokuapp.com/boards/Semret/cards?text=${newCard.card.text}&emoji=${newCard.card.emoji}`)
    .then((response) => {
      this.setState({
        cards: updateCards
      });
    })
    .catch((error) => {
      this.setState({
        error: error.message
      });
    })
  }

  render() {
    const emoji = require("emoji-dictionary");

    const messages = this.state.cards

    const messageCollection = messages.map((message, i) => {
      return <Card
        key={i}
        id={message.card.id}
        text={message.card.text}
        emoji={emoji.getUnicode(`${message.card.emoji}`)}
        deleteCardCallback={this.deleteThisCard}
        />
    });
    return (
      <div>
        <section className="form-section">
          <NewCardForm
            addNewCardToCollection={this.addNewCard}
            />
        </section>
        <section className="board">
          {messageCollection}
        </section>
      </div>
    )
  }
}

Board.propTypes = {
  cards: PropTypes.array,
};

export default Board;
