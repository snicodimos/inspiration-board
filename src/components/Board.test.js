import React from 'react';
import Board from './Board';
import { shallow } from 'enzyme';


describe('Board', () => {
  test('that it matches an existing snapshot for deleting', () => {
    const wrapper = shallow( <Board deleteThisCard={() => {} } />);
    expect(wrapper).toMatchSnapshot();
  });

  test('that it matches an existing snapshot for adding', () => {
    const wrapper = shallow( <Board addNewCard={() => {} } />);
    expect(wrapper).toMatchSnapshot();
  });
});
