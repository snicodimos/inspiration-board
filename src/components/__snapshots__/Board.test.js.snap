// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board that it matches an existing snapshot for adding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    addNewCard={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <NewCardForm
          addNewCardToCollection={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addNewCardToCollection": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <NewCardForm
            addNewCardToCollection={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addNewCardToCollection": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Board that it matches an existing snapshot for deleting 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    deleteThisCard={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <NewCardForm
          addNewCardToCollection={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addNewCardToCollection": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <NewCardForm
            addNewCardToCollection={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addNewCardToCollection": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
